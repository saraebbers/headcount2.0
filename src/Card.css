.above {
  color: green;
}

.below {
  color: red;
}

.card {
  border: solid 2px black;
  color: black;
}

ul {
list-style: none
}