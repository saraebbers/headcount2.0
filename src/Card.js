import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css'

const Card = ( {location, stats} ) => {
    const displayStats = Object.keys(stats).map(stat => {
      return ( <li>{stat} {stats[stat] }</li>
        )
    })
    
    return(
      <div className='card'>
        <h4>{ location }</h4>
        <button>Select</button>
        <ul>
          {displayStats} 
        </ul>
      </div>
    )
  }

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
}

export default Card;