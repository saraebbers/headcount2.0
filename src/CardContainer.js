import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './CardContainer.css';

const CardContainer = ( {displayData} ) => {

  const cards = displayData.map(district => {
    return <Card 
            key={Date.now()} 
            {...district} />
  })

  return(
    <div>
      <h3 className='cardContainerHeading'>Data for Each District Contained Here</h3>
      <section className='cardContainer-grid'>
        { cards }
      </section>
    </div>
  )
}

CardContainer.propTypes = {
  displayData: PropTypes.array.isRequired
}

export default CardContainer;