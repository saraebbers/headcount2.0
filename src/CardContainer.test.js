import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';
import Card from './Card';

describe('CardContainer', () => {
  it('matches the snapshot', () => {
    let mockDisplayData = 
    [{location: 'NorthCarolina', 
    stats: {1999: 0.007, 2099: 1.999 }}, 
    {location: 'VIRGINIA', 
    stats: {1909: 0.107, 3999: 5.999 }}];
    let wrapper = shallow(<CardContainer displayData={mockDisplayData}/>);

    expect(wrapper).toMatchSnapshot()
  });

  it('renders all the cards', () => {
    let mockDisplayData = 
    [ {location: 'NorthCarolina', 
    stats: {1999: 0.007, 2099: 1.999 }}, 
    {location: 'VIRGINIA', 
    stats: {1909: 0.107, 3999: 5.999 }}];

    let wrapper = shallow(<CardContainer
        displayData={ mockDisplayData }
          />)
    expect(wrapper.find(Card).length).toEqual(2)
  })

})