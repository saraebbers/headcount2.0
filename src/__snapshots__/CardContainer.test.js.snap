// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    displayData={
        Array [
            Object {
              "location": "NorthCarolina",
              "stats": Object {
                "1999": 0.007,
                "2099": 1.999,
              },
            },
            Object {
              "location": "VIRGINIA",
              "stats": Object {
                "1909": 0.107,
                "3999": 5.999,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="cardContainerHeading"
>
          Data for Each District Contained Here
</h3>,
        <section
          className="cardContainer-grid"
>
          <Card
                    location="NorthCarolina"
                    stats={
                              Object {
                                        "1999": 0.007,
                                        "2099": 1.999,
                                      }
                    }
          />
          <Card
                    location="VIRGINIA"
                    stats={
                              Object {
                                        "1909": 0.107,
                                        "3999": 5.999,
                                      }
                    }
          />
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Data for Each District Contained Here",
          "className": "cardContainerHeading",
        },
        "ref": null,
        "rendered": "Data for Each District Contained Here",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              location="NorthCarolina"
              stats={
                            Object {
                                          "1999": 0.007,
                                          "2099": 1.999,
                                        }
              }
/>,
            <Card
              location="VIRGINIA"
              stats={
                            Object {
                                          "1909": 0.107,
                                          "3999": 5.999,
                                        }
              }
/>,
          ],
          "className": "cardContainer-grid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1543612466667",
            "nodeType": "function",
            "props": Object {
              "location": "NorthCarolina",
              "stats": Object {
                "1999": 0.007,
                "2099": 1.999,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1543612466667",
            "nodeType": "function",
            "props": Object {
              "location": "VIRGINIA",
              "stats": Object {
                "1909": 0.107,
                "3999": 5.999,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="cardContainerHeading"
>
            Data for Each District Contained Here
</h3>,
          <section
            className="cardContainer-grid"
>
            <Card
                        location="NorthCarolina"
                        stats={
                                    Object {
                                                "1999": 0.007,
                                                "2099": 1.999,
                                              }
                        }
            />
            <Card
                        location="VIRGINIA"
                        stats={
                                    Object {
                                                "1909": 0.107,
                                                "3999": 5.999,
                                              }
                        }
            />
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Data for Each District Contained Here",
            "className": "cardContainerHeading",
          },
          "ref": null,
          "rendered": "Data for Each District Contained Here",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                location="NorthCarolina"
                stats={
                                Object {
                                                "1999": 0.007,
                                                "2099": 1.999,
                                              }
                }
/>,
              <Card
                location="VIRGINIA"
                stats={
                                Object {
                                                "1909": 0.107,
                                                "3999": 5.999,
                                              }
                }
/>,
            ],
            "className": "cardContainer-grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1543612466667",
              "nodeType": "function",
              "props": Object {
                "location": "NorthCarolina",
                "stats": Object {
                  "1999": 0.007,
                  "2099": 1.999,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1543612466667",
              "nodeType": "function",
              "props": Object {
                "location": "VIRGINIA",
                "stats": Object {
                  "1909": 0.107,
                  "3999": 5.999,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
